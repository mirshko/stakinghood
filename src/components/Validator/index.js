import React from "react";
import { Box as RebassBox, Flex as RebassFlex, Text } from "rebass";
import styled from "styled-components";
import PropTypes from "prop-types";
import Tag from "../Tag";
import UnstyledButton from "../UnstyledButton";

const ValidatorIcon = styled(RebassBox)`
  height: 40px;
  width: 40px;
  border-radius: calc(40px / 2);
  display: flex;
  align-items: center;
  justify-content: center;
  user-select: none;
  font-size: 24px;
`;

const formatter = num => (num > 999 ? (num / 1000).toFixed(1) + "K" : num);

const Flex = styled(RebassFlex)`
  ${props =>
    props.bottomBorder ? `border-bottom: 1px solid var(--brand-light);` : null}
`;

const Validator = ({
  symbol,
  name,
  color,
  emoji,
  totalStake,
  onClick,
  bondedStake,
  uptime,
  bottomBorder
}) => (
  <UnstyledButton onClick={onClick}>
    <Flex
      bottomBorder={bottomBorder}
      pt={20}
      pb={20 - 1}
      justifyContent="space-between"
      alignItems="center"
    >
      <RebassFlex>
        <ValidatorIcon bg={color}>{emoji}</ValidatorIcon>
        <RebassBox ml="8px">
          <Text
            title={name}
            lineHeight="24px"
            fontSize="16px"
            fontWeight="var(--weight-medium)"
          >
            {symbol}
          </Text>
          <Text lineHeight="12px" fontSize="12px">
            {formatter(totalStake)} ATOM
          </Text>
        </RebassBox>
      </RebassFlex>
      <Tag tall>
        {bondedStake ? `${formatter(bondedStake)} ATOM` : `${uptime}%`}
      </Tag>
    </Flex>
  </UnstyledButton>
);

Validator.defaultProps = {
  color: "#000",
  emoji: "ðŸ“Ÿ",
  bottomBorder: true
};

Validator.propTypes = {
  symbol: PropTypes.string.isRequired,
  name: PropTypes.string,
  color: PropTypes.string,
  emoji: PropTypes.string,
  totalStake: PropTypes.number.isRequired,
  bondedStake: PropTypes.number.isRequired,
  onClick: PropTypes.func.isRequired,
  uptime: PropTypes.number,
  bottomBorder: PropTypes.bool
};

export default Validator;
