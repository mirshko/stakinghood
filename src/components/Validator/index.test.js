import React from "react";
import { shallow } from "enzyme";

import Validator from './index';

describe("<Validator />", () => {
  it("it works", () => {
    const wrapper = shallow(<Validator
      onClick={() => console.log("Clicked")}
      symbol="MYTH"
      name="Mythos Capital"
      color="#000"
      emoji="ðŸ”’"
      totalStake={365575}
      bondedStake={15575}
      uptime={100.00}
    />);
    expect(wrapper).toMatchSnapshot();
  });
});
