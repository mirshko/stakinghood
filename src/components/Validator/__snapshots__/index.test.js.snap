// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Validator /> it works 1`] = `
<UnstyledButton
  onClick={[Function]}
>
  <Styled(Styled(styled.div))
    alignItems="center"
    bottomBorder={true}
    justifyContent="space-between"
    pb={19}
    pt={20}
  >
    <Styled(styled.div)>
      <Styled(styled.div)
        bg="#000"
      >
        ðŸ”’
      </Styled(styled.div)>
      <styled.div
        ml="8px"
      >
        <Styled(styled.div)
          fontSize="16px"
          fontWeight="var(--weight-medium)"
          lineHeight="24px"
          title="Mythos Capital"
        >
          MYTH
        </Styled(styled.div)>
        <Styled(styled.div)
          fontSize="12px"
          lineHeight="12px"
        >
          365.6K
           ATOM
        </Styled(styled.div)>
      </styled.div>
    </Styled(styled.div)>
    <Styled(Styled(styled.div))
      tall={true}
    >
      15.6K ATOM
    </Styled(Styled(styled.div))>
  </Styled(Styled(styled.div))>
</UnstyledButton>
`;
